crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -258,7 +258,7 @@ pub(super) struct TypeInferenceBuilder<'db, 'ast> {
     /// ```
     ///
     /// [`check_overloaded_functions`]: TypeInferenceBuilder::check_overloaded_functions
-    called_functions: FxHashSet<FunctionType<'db>>,
+    called_functions: FxOrderSet<FunctionType<'db>>,
```

Comment:
Nit: Could we use an `FxIndexSet`. I believe insertion order should be "good enough."